code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -350,22 +353,30 @@ mob/living/simple_animal/hostile/hitby(atom/movable/AM as mob|obj,var/speed = TH
 //////////////////////////////
 
 /mob/living/simple_animal/hostile/proc/validator_living(var/mob/living/L, var/atom/current)
-	if(ismech(L))
-		var/mob/living/heavy_vehicle/M = L
-		if(!M.pilots?.len)
-			return FALSE
-
+	if(L.health <= 0 || L.stat == DEAD)
+		return FALSE
 	if((L.faction == src.faction) && !attack_same)
 		return FALSE
 	if(L in friends)
 		return FALSE
-	if(!L.stat)
-		var/current_health = INFINITY
-		if (isliving(current))
-			var/mob/living/M = current
-			current_health = M.health
-		if(L.health < current_health)
+
+	if(length(valid_living_targets)) // it has a specific list of targets
+		if(is_type_in_list(L, valid_living_targets))
 			return TRUE
+		return FALSE
+	else
+		if(ismech(L))
+			var/mob/living/heavy_vehicle/M = L
+			if(!M.pilots?.len)
+				return FALSE
```

Comment:
Why did you pull this all the way down? Having it at the top makes it skip some calculations, which is a good idea for AI.